### PR TITLE
feat(auth): support authentication

### PR DESCRIPTION
**Pull request by knqyf263**
If `registry.Auth` is specified, the registry requests authentication.

### CODE REVIEW

**Comment by simar7** [comment]
Not related to this change but where did some of the structs like `DockerImageDetail` and `DockerRegistryStore` go? as a result some of the tests are not compiling anymore. Would you know @knqyf263? Maybe they got lost somehow in the last merge?

**Comment by knqyf263** [comment]
Ah, I didn't know there are tests. After refactoring, it got to be unnecessary, so I removed it. I'll fix tests.

**Comment by knqyf263** [comment]
@simar7 I'm sorry, but I don't understand why they are necessary so that I removed them. If they are necessary, let me know, please.
https://github.com/aquasecurity/testdocker/blob/deab5f23ad039944731ef8892195ca8fae6767f1/registry/server_test.go#L207-L229

**Comment by simar7** [comment]
> @simar7 I'm sorry, but I don't understand why they are necessary so that I removed them. If they are necessary, let me know, please.
> 
> https://github.com/aquasecurity/testdocker/blob/deab5f23ad039944731ef8892195ca8fae6767f1/registry/server_test.go#L207-L229

Looks good now!

File: auth/router.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	if s[0] != a.auth.User || s[1] != a.auth.Password {
+		return errdefs.Unauthorized(xerrors.New("invalid username/password"))
+	}
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"iss": issuer,
+	})
+
+	tokenString, err := token.SignedString([]byte(a.auth.Secret))
+	if err != nil {
+		return err
```
Comment (simar7) [inline auth/router.go:99]: What do you think about this?
```suggestion
		return errdefs.Unavailable(err)
```

**Comment by simar7** [comment]
Oh one more thing, could you also update the Readme with a check for auth? https://github.com/aquasecurity/testdocker/blob/master/README.md

File: auth/router.go (hunk 1 above)
Comment (knqyf263) [reply auth/router.go]: Nica catch! Thanks.
https://github.com/aquasecurity/testdocker/pull/2/commits/e9b63fc782c4f77bbd544d6a6642b25ab198b1f4

**Pull request merged**